### PR TITLE
fix(Core/Spells): cyclone shouldn't be casted on totems

### PR DESCRIPTION
**Pull request by ErnestoGonzalezC**
## Changes Proposed:
-  Cyclone shouldn't be casted on totems.

## Issues Addressed:
- Closes 

## Tests Performed:

- Compile without errors on Windows.

## How to Test the Changes:

- Login with a druid and shaman, try throwing a cyclone on a totem and you will see that it now says the totem is immune.

## Known Issues and TODO List:

- [ ]
- [ ] 

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Sedatyf** [comment]
Hey @ErnestoGonzalezC 
I tested your changes, and it works like a charm (please see video below). However, do you have any documentation about that? In my opinion, I agree with you, Cyclone shouldn't work on totems, but I couldn't find any information about it.

https://user-images.githubusercontent.com/59012307/115087956-820abe00-9f0f-11eb-8d75-1eaab97ab98b.mp4

**Comment by ErnestoGonzalezC** [comment]
@Sedatyf A few days ago another person asked me if I have proof of something, I will answer you just like him. In my country, the internet is almost inaccessible and extremely expensive, so the corrections I make are based on my experience as a player in Warmane, so I cannot afford to investigate, my apologies, but I cannot.

**Comment by Sedatyf** [comment]
@ErnestoGonzalezC no worries, I'll check this out.

**Comment by Sedatyf** [comment]
I couldn't find any informations about it. But I agree with Ernesto, it should work like this.

File: src/server/game/Entities/Totem/Totem.cpp [hunk 1]
```diff
@@ -152,6 +152,10 @@ bool Totem::IsImmunedToSpellEffect(SpellInfo const* spellInfo, uint32 index) con
             spellInfo->Effects[index].TargetA.GetCheckType() != TARGET_CHECK_ENTRY && spellInfo->Id != 55277 && spellInfo->Id != 6277)
         return true;
 
+    // Cyclone shouldn't be casted on totems
+    if (spellInfo->Id == 33786)
```
Comment (Kitzunu) [inline src/server/game/Entities/Totem/Totem.cpp:156]: At least don't hardcode the ID, please make a define, perhaps in totem.h

**Comment by ErnestoGonzalezC** [comment]
@Kitzunu Done.

File: src/server/game/Entities/Totem/Totem.h [hunk 2]
```diff
@@ -21,6 +21,8 @@ enum TotemType
 #define SENTRY_TOTEM_ENTRY    3968
 #define EARTHBIND_TOTEM_ENTRY 2630
 
+#define IMMUNE_TOTEM  33786
```
Comment (Kitzunu) [inline src/server/game/Entities/Totem/Totem.h:24]:
```suggestion
#define SPELL_CYCLONE 33786
```

File: src/server/game/Entities/Totem/Totem.cpp [hunk 3]
```diff
@@ -152,6 +152,10 @@ bool Totem::IsImmunedToSpellEffect(SpellInfo const* spellInfo, uint32 index) con
             spellInfo->Effects[index].TargetA.GetCheckType() != TARGET_CHECK_ENTRY && spellInfo->Id != 55277 && spellInfo->Id != 6277)
         return true;
 
+    // Cyclone shouldn't be casted on totems
+    if (IMMUNE_TOTEM)
```
Comment (Kitzunu) [inline src/server/game/Entities/Totem/Totem.cpp:156]:
```suggestion
    if (spellInfo->Id == SPELL_CYCLONE)
```

**Comment by FrancescoBorzi** [comment]
@ErnestoGonzalezC gracias camarada!

@Sedatyf thanks for testing this!

I think it's safe to merge this. If we ever find out that this change was not correct, we can always easily revert it as it's just a few lines of code.

**Comment by Sedatyf** [comment]
> @ErnestoGonzalezC gracias camarada!
> 
> @Sedatyf thanks for testing this!
> 
> I think it's safe to merge this. If we ever find out that this change was not correct, we can always easily revert it as it's just a few lines of code.

If you say so! And you're welcome, it was easy to test 😄

**Comment by pklloveyou** [comment]
Tested

**Comment by Sedatyf** [comment]
It was already tested. It just need to be merged

File: src/server/game/Entities/Totem/Totem.h [hunk 4]
```diff
@@ -21,6 +21,8 @@ enum TotemType
 #define SENTRY_TOTEM_ENTRY    3968
 #define EARTHBIND_TOTEM_ENTRY 2630
 
+#define SPELL_CYCLONE 33786
```
Comment (Winfidonarleyan) [inline src/server/game/Entities/Totem/Totem.h:24]:
```cpp
constexpr uint32 SPELL_CYCLONE = 33786;
```

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**